cEP-0016.md

```diff
@@ -0,0 +1,455 @@
+   disable CI builds.
+2. Each type of commit is then handled as a separate case.
+3. For GitHub Pull Request temporary merge commits:
+   - As a solution to this problem, settings will be added in `GitCommitBear`
+     that would identify GitHub Pull Request temporary merge commits and ignore
+     them.
+   - `GitCommitBear` would identify and check contents of the unmerged parent of
+     these merge commits.
+4. A meta bear called `VCSCommitBear` would be created that would return a
+   `HiddenResult` comprising of a tuple that will contain information about
+   the type of git commit, commit sha and lists of modified files, added and
+   deleted files.
+   This meta bear would be used to inspect each type of special commit.
+5. For commits that disable CI builds:
+   This would involve making a new bear named `CISkipInspectBear` to inspect
+   commits that disable CI builds. This bear would contain metadata about
+   various CI providers that are supported by coala such as Circle CI,
+   Travis CI, Semaphore CI etc. Most of the CI providers use sequences like
+   `[ci skip]` and `[skip ci]` for skipping CI build on commits. CI build
+   don't occur for commits that have either of these sequences anywhere in
+   the commit message.
+   This bear would inspect the changes made in such commits and verify that they
+   can skip CI build that is with respect to modified files and whether the
+   corresponding CI engine allows skipping build or not. An appropriate message
+   is returned if skipping build is not supported.
+   - If the commit type from the output of `VCSCommitBear` indicates that the
+     commit skips CI build, inspection is done inside `CISkipInspectBear`
+     otherwise return.
+   - Files modified in the commit should include files that are not covered by
+     `.coafile` or tests. If this is not the case, an error is raised to exclude
+     `[ci skip]` or `[skip ci]` from the commit message.
+6. For merge commits:
+   - If the commit type from the output of `VCSCommitBear` indicates that the
+     commit is a `git merge` commit, inspection is done in `GitMergeInspectBear`
+     otherwise return.
+   - `GitMergeInspectBear` will be used by find if the commit tries to merge
+     master or any other branch onto the pull request, message is displayed to
+     perform a `git rebase` instead.
+7. For revert commits:
+   - If the commit type from the output of `VCSCommitBear` indicates that the
+     commit is a `git revert` commit, inspection is done in
+     `GitRevertInspectBear` otherwise return.
+   - In `GitRevertInspectBear`, the patches of both commits are compared to
+     ensure that they are in fact reverse of each other. Minor differences are
+     allowed.
+   - In case of total mismatch, a note in the commit message highlighting that
+     it is not a clean revert is expected.
+   - A message suggesting improvements in the commit message is displayed.
+
+## Code Prototypes
```

Comment:
I think we might want to get rid of this section since it changes everytime the PR for VCSCommitBear updated, and it blocks this cEP to get merged faster. The explanation above is good enough imo :)